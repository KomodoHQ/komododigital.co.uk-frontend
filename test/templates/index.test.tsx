import React from 'react';
import Index from '../../src/templates/index';
import renderer from 'react-test-renderer';
import GatsbyData from '../../gatsby-config';

test('Home template is rendered', () => {

  const htmlAst = {
    type: 'element',
    tagName: 'span',
    properties: {},
    children: [
      {
        type: 'text',
        value: 'Lorem Ipsum is the industry standard typesetting text',
      },
    ],
  };

  const hocProps = {
    services: [],
    caseStudies: [],
    insights: [],
    subtitle: 'subtitle',
    title: 'title',
    intro: htmlAst,
    aboutUsIntro: htmlAst,
    approachIntro: htmlAst,
    caseStudiesIntro: htmlAst,
    clientPortfoliosIntro: htmlAst,
    insightsIntro: htmlAst,
    contactsIntro: htmlAst,
    data: {
      site: GatsbyData,
      logo: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      },
      logo_inverted: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      },
      avatar: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      },
      client_cop: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_digitalbarriers: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_elanders: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_gichd: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_htl: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_isg: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_itv: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_legrand: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_spacegroup: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_newcastleuni: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_northumbriapolice: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_northumbriauni: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_onward: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_orchard: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_streetstream: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } },
      client_thingco: { childImageSharp: { fixed: {width: 1, srcSet:'',src:'',height:1} } }

    },
  };

  const component = renderer.create(<Index {...hocProps} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
