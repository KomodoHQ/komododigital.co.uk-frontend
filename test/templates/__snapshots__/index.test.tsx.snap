// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home template is rendered 1`] = `
Array [
  <div
    className="navigationBar"
  >
    <h1>
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Komodo
      </a>
    </h1>
    <ul>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/case-studies"
          onClick={[Function]}
        >
          Case Studies
        </a>
      </li>
      <li>
        <a
          href="/blog-list"
          onClick={[Function]}
        >
          Insights
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </div>,
  <noscript>
    You have Javascript disabled. While it isn't needed, it will make your experience nicer.
  </noscript>,
  <div
    style={
      Object {
        "paddingTop": "80px",
      }
    }
  >
    <div
      className="cs-wrapper invert"
      style={
        Object {
          "backgroundColor": "#000000",
          "paddingBottom": "200px",
          "paddingTop": "200px",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <span>
          subtitle
        </span>
        <h2>
          title
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="cs-wrapper "
      style={
        Object {
          "backgroundColor": "inherit",
          "paddingBottom": "100px",
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          About Us
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="csi-wrapper"
    >
      <img
        alt="Komodo at work"
        src={Object {}}
      />
    </div>
    <div
      className="cs-wrapper "
      style={
        Object {
          "backgroundColor": "inherit",
          "paddingBottom": "100px",
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          Approach
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="QuadGrid"
    />
    <div
      className="cs-wrapper "
      style={
        Object {
          "backgroundColor": "inherit",
          "paddingBottom": "100px",
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          Case Studies
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="seemorebutton-wrapper"
    >
      <a
        href="/About"
        onClick={[Function]}
      >
        <button>
          See More Work
        </button>
      </a>
    </div>
    <div
      className="cs-wrapper "
      style={
        Object {
          "backgroundColor": "inherit",
          "paddingBottom": "100px",
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          Client Portfolio
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="PortfolioGrid"
    >
      <span />
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
      <div
        className="StandardItem"
      >
        <a
          href="/About"
          onClick={[Function]}
        >
          <img
            src={Object {}}
          />
        </a>
      </div>
    </div>
    <div
      className="cs-wrapper "
      style={
        Object {
          "backgroundColor": "inherit",
          "paddingBottom": "100px",
          "paddingTop": "100px",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          Insights
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="BlogGrid"
    />
    <div
      className="seemorebutton-wrapper"
    >
      <a
        href="/About"
        onClick={[Function]}
      >
        <button>
          See More Insights
        </button>
      </a>
    </div>
    <div
      className="cs-wrapper "
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "paddingBottom": "50px",
          "paddingTop": "50px",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          TITLE HERE
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="contactsection-wrapper"
    >
      <div
        className="contactsection"
      >
        <div
          className="vcard"
        >
          <img
            height="70"
            src={Object {}}
            width="70"
          />
          <div>
            <h1>
              Armin Talic
            </h1>
            <h2>
              Commercial Director
            </h2>
            <a
              href="mailto: {email}"
            >
              story@vcard.com
            </a>
          </div>
        </div>
        <div
          className="vcard"
        >
          <img
            height="70"
            src={Object {}}
            width="70"
          />
          <div>
            <h1>
              Matt Moran
            </h1>
            <h2>
              Director
            </h2>
            <a
              href="mailto: {email}"
            >
              story@vcard.com
            </a>
          </div>
        </div>
        <div
          className="contactform"
        >
          <form
            action="https://formspree.io/hello@komododigital.co.uk"
            method="POST"
          >
            <label
              htmlFor="form-name"
            >
              Your Name
              <input
                id="form-name"
                name="name"
                placeholder="Joe Blogs"
                type="text"
              />
            </label>
            <label
              htmlFor="form-email"
            >
              Your Email
              <input
                id="form-email"
                name="email"
                placeholder="joe@blogs.com"
                type="email"
              />
            </label>
            <label
              htmlFor="form-service"
            >
              Service
              <select
                id="form-service"
                name="service"
              >
                <option>
                  Choose Service
                </option>
                <option
                  value="Discovery"
                >
                  Discovery
                </option>
                <option
                  value="Design"
                >
                  Design
                </option>
                <option
                  value="Development"
                >
                  Development
                </option>
                <option
                  value="Delivery"
                >
                  Delivery
                </option>
              </select>
            </label>
            <label
              htmlFor="form-sector"
            >
              Sector
              <select
                id="form-sector"
                name="sector"
              >
                <option>
                  Choose Sector
                </option>
                <option
                  value="Education"
                >
                  Education
                </option>
                <option
                  value="Law Enforcement"
                >
                  Law Enforcement
                </option>
                <option
                  value="Startups"
                >
                  Startups
                </option>
                <option
                  value="Social Housing"
                >
                  Social Housing
                </option>
              </select>
            </label>
            <label
              htmlFor="form-message"
            >
              Your Message
              <textarea
                cols={30}
                id="form-message"
                name="message"
                rows={10}
              />
            </label>
            <div>
              <input
                type="submit"
                value="Send"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
    <footer>
      <div
        className="komodo"
      >
        Komodo
      </div>
      <div
        className="address"
      >
        <span>
          Address
        </span>
        <address>
          Komodo Digital,
          <br />
          63 Westgate Rd,
          <br />
          Newcastle upon Tyne,
          <br />
          NE1 1SG
        </address>
        <address>
          Komodo | The Core
          <br />
          Science Central, Bath Ln,
          <br />
          Newcastle upon Tyne
          <br />
          NE4 5TF
        </address>
      </div>
      <div
        className="sitemap"
      >
        <span>
          Sitemap
        </span>
        <ul>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Case Studies
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Insights
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Careers
            </a>
          </li>
        </ul>
        <ul>
          <li>
            <a
              aria-current="page"
              href="/"
              onClick={[Function]}
            >
              Contact
            </a>
          </li>
          <li>
            <a
              href="https://twitter.com/komodohq"
            >
              Twitter
            </a>
          </li>
          <li>
            <a
              href="https://facebook.com/komodohq"
            >
              Facebook
            </a>
          </li>
          <li>
            <a
              href="https://linkedin.com/komodohq"
            >
              LinkedIn
            </a>
          </li>
        </ul>
      </div>
    </footer>
    <footer
      className="legal-footer"
    >
      <div
        className="company"
      >
        <span>
          Komodo Digital Copyright © 2018
        </span>
        <br />
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          Privacy Policy
        </a>
         | 
        <a
          aria-current="page"
          href="/"
          onClick={[Function]}
        >
          Terms and Conditions
        </a>
      </div>
      <div
        className="contact"
      >
        Get in touch on:
        <br />
        <a
          href="/tel:0191 228 6555"
          onClick={[Function]}
        >
          0191 228 6555
        </a>
      </div>
    </footer>
  </div>,
  <noscript>
    <img
      alt=""
      height="1"
      src="https://dc.ads.linkedin.com/collect/?pid=377068&fmt=gif"
      width="1"
    />
  </noscript>,
]
`;
